jubilant/_juju.py

```diff
@@ -837,6 +842,79 @@ def scp(
 
         self.cli(*args)
 
+    def secrets(self) -> list[Secret[Hidden]]:
+        """Get all secrets in the model.
+
+        Returns:
+            A list of :class:`Secret[Hidden]` objects, one for each secret in the model.
+        """
+        stdout = self.cli('secrets', '--format', 'json')
+        output = json.loads(stdout)
+        return [
+            Secret[Hidden]._from_dict({'uri': uri_from_juju, **obj})
+            for uri_from_juju, obj in output.items()
+        ]
+
+    @overload
+    def show_secret(
+        self,
+        identifier: str | SecretURI,
+        *,
+        reveal: Literal[True],
+        revision: int | None = None,
+        revisions: Literal[False] = False,
+    ) -> Secret[Revealed]: ...
+
+    @overload
+    def show_secret(
+        self,
+        identifier: str | SecretURI,
+        *,
+        reveal: Literal[False] = False,
+        revision: int | None = None,
+        revisions: Literal[False] = False,
+    ) -> Secret[Hidden]: ...
+
+    @overload
+    def show_secret(
+        self,
+        identifier: str | SecretURI,
+        *,
+        reveal: Literal[False] = False,
+        revision: None = None,
+        revisions: Literal[True],
+    ) -> Secret[Hidden]: ...
+
+    def show_secret(
+        self,
+        identifier: str | SecretURI,
+        *,
+        reveal: bool = False,
+        revision: int | None = None,
+        revisions: bool = False,
+    ) -> Secret[Revealed] | Secret[Hidden]:
+        """Get the content of a secret.
+
+        Args:
+            identifier: Name or URI of the secret to reveal.
```

Comment:
Because we're only "reveal"ing it if reveal=True, let's change this to:

```suggestion
            identifier: Name or URI of the secret to return.
```

Reply:
done